lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -3558,6 +3559,13 @@ instance IsServerError ErrSelectAssets where
                         , "must specify enough ada. Here are the problematic "
                         , "outputs:\n" <> pretty (indentF 2 $ blockListF xs)
                         ]
+                OutputsInsufficient e ->
+                    apiError err403 MintedNotSpent $ mconcat
```

Comment:
Isn't that a server error or do we expect clients to specify all the necessary outputs? I am asking because of the PR description which mentions a modification of the change generation to include minted value and I see arguments for both.

Reply:
> do we expect clients to specify all the necessary outputs?

Yes. The current design expects all minted tokens to be spent in user-specified outputs.

But you're quite right, we could adjust the design to return minted-but-not-spent tokens back as change.

In my opinion, this is a bit of a gap in the requirements. See: https://github.com/input-output-hk/cardano-wallet/pull/2725#discussion_r655894252

Reply:
Yeah, I'll have to chase up the requirements from this.

Reply:
For now, I think it's best to leave this as-is: require the user to explicitly send the tokens to an address. The requirements are a bit unclear on this point, and to allow users to not specify an address would require more work. Some parts of the code don't care at all (change bundle generation), but the selection algorithm does.

The API rejects requests to mint that don't also include an address to send them, but I've added an error the server could return if the API were to allow it, just in-case the behavior of the API is changed in future.